gammapy/maps/hpxnd.py

```diff
@@ -648,3 +649,28 @@ def get_angle(x, t):
         ax.coords.grid(color='w', linestyle=':', linewidth=0.5)
 
         return fig, ax, p
+
+    def make_region_mask(self, region, inside=True):
+        """Create a mask of a given region
+
+        TODO: implement list of region for each axis
```

Comment:
Concerning this TODO:

How should we handle extra axes?

I'm -1 on trying to define ND regions, that's a huge project.

That leaves the option of making a 2D mask, or the same 2D mask duplicated along the axes.

Here we did the replicated along the energy axis:
http://docs.gammapy.org/dev/_modules/gammapy/cube/core.html#SkyCube.region_mask
I think for `solid_angle` that's also what you did @registerrier ?
I would be in favour of returning 2D by default, but adding a bool option which replicates and makes an ND mask.

And then I'd also suggest to add an example how to do something more fancy, e.g. if there is an energy-dependent FOV one can make an empty mask, then call `iter_by_image` and fill as you like:
http://docs.gammapy.org/dev/maps/index.html#iterating-on-a-map
Inventing an API for that is difficult, and things can be added later, so for now I'd suggest to solve this via docs.

Reply:
Well that was the idea basically. 
You can provide a list/tuple of SkyRegions and then apply iteration.